cpp_filter/filter.rs

```diff
@@ -0,0 +1,144 @@
+mod rpc;
```

Comment:
These files should be in a different directory, since they are unrelated to the cpp filter for Envoy.

Reply:
I suggest `sim_filter/filter.rs`.

Reply:
That makes sense.  I think I will delete them for now since they are just examples, and we already have examples in the tracing_sim project.